e2etest/declarativeRunner.go

```diff
@@ -73,44 +138,54 @@ func RunScenarios(
+			credentialTypes := getValidCredCombinationsForFromTo(fromTo, requestedCredentialTypesSrc, requestedCredentialTypesDst)
 
-			// Sub-test name is not globally unique (it doesn't need to be) but it is more human-readable
-			subtestName := fmt.Sprintf("%s-%s", op, fromTo)
-			if scenarioSuffix != "" {
-				subtestName += "-" + scenarioSuffix
-			}
+			scenarioList := make([]scenario, 0)
 
-			hsToUse := hooks{}
-			if hs != nil {
-				hsToUse = *hs
-			}
+			for _, credTypes := range credentialTypes {
+				// Create unique name for generating container names
+				compactScenarioName := fmt.Sprintf("%.4s-%s-%c-%c%c", suiteName, testName, op.String()[0], fromTo.From().String()[0], fromTo.To().String()[0])
+				fullScenarioName := fmt.Sprintf("%s.%s.%s-%s", suiteName, testName, op.String(), fromTo.String())
+				// Sub-test name is not globally unique (it doesn't need to be) but it is more human-readable
+				subtestName := fmt.Sprintf("%s-%s", op, fromTo)
 
-			// Handles destination being different account type
-			isSourceAcc := true
-			if destAccountType != accountType {
-				isSourceAcc = false
-			}
-			s := scenario{
-				srcAccountType:      accountType,
-				destAccountType:     destAccountType,
-				subtestName:         subtestName,
-				compactScenarioName: compactScenarioName,
-				fullScenarioName:    fullScenarioName,
-				operation:           op,
-				fromTo:              fromTo,
-				validate:            validate,
-				p:                   p, // copies them, because they are a struct. This is what we need, since they may be morphed while running
-				hs:                  hsToUse,
-				fs:                  fs.DeepCopy(),
-				needResume:          operations&eOperation.Resume() != 0,
-				stripTopDir:         p.stripTopDir,
-				isSourceAcc:         isSourceAcc,
+				hsToUse := hooks{}
+				if hs != nil {
+					hsToUse = *hs
+				}
+
+				if scenarioSuffix != "" {
+					subtestName += "-" + scenarioSuffix
+				}
+
+				// Handles destination being different account type
+				isSourceAcc := true
+				if destAccountType != accountType {
+					isSourceAcc = false
+				}
+
+				s := scenario{
+					srcAccountType:      accountType,
+					destAccountType:     destAccountType,
+					subtestName:         subtestName,
+					compactScenarioName: compactScenarioName,
+					fullScenarioName:    fullScenarioName,
+					operation:           op,
+					fromTo:              fromTo,
+					credTypes:           credTypes,
+					validate:            validate,
+					p:                   p, // copies them, because they are a struct. This is what we need, since they may be morphed while running
+					hs:                  hsToUse,
+					fs:                  fs.DeepCopy(),
+					needResume:          operations&eOperation.Resume() != 0,
+					stripTopDir:         p.stripTopDir,
+					isSourceAcc:         isSourceAcc,
+				}
+
+				scenarioList = append(scenarioList, s)
```

Comment:
Nit: Append to scenarios only and not create a 2D array, maybe? There's no grouping logic required when we're running all tests in parallel.

Reply:
Or have few comments explaining what 2D array of scenarios contain.